test/passes/simplify-globals_enable-mutable-globals.txt

```diff
@@ -110,3 +110,104 @@
   )
  )
 )
+(module
+ (type $FUNCSIG$ii (func (param i32) (result i32)))
+ (global $g1 (mut i32) (i32.const 1))
+ (global $g2 (mut i32) (i32.const 1))
+ (func $f (; 0 ;) (type $FUNCSIG$ii) (param $x i32) (result i32)
+  (global.set $g1
+   (i32.const 100)
+  )
+  (global.set $g2
+   (local.get $x)
+  )
+  (if
+   (local.get $x)
+   (return
+    (i32.const 0)
+   )
+  )
+  (local.set $x
+   (i32.add
+    (global.get $g1)
+    (global.get $g2)
+   )
+  )
+  (if
+   (local.get $x)
+   (return
+    (i32.const 1)
+   )
+  )
+  (global.set $g1
+   (i32.const 200)
+  )
+  (global.set $g2
+   (local.get $x)
+  )
+  (local.set $x
+   (i32.add
+    (i32.const 200)
+    (global.get $g2)
+   )
+  )
+  (local.get $x)
+ )
+)
+(module
+ (type $FUNCSIG$ii (func (param i32) (result i32)))
+ (global $g1 (mut i32) (i32.const 1))
+ (global $g2 (mut i32) (i32.const 1))
+ (func $f (; 0 ;) (type $FUNCSIG$ii) (param $x i32) (result i32)
+  (global.set $g1
+   (i32.const 100)
+  )
+  (global.set $g2
+   (local.get $x)
+  )
+  (local.set $x
+   (i32.add
+    (i32.add
+     (i32.const 100)
+     (i32.const 100)
+    )
+    (global.get $g2)
+   )
+  )
+  (local.get $x)
+ )
+)
+(module
+ (type $FUNCSIG$ii (func (param i32) (result i32)))
+ (global $g1 (mut i32) (i32.const 1))
+ (global $g2 (mut i32) (i32.const 1))
+ (func $no (; 0 ;) (type $FUNCSIG$ii) (param $x i32) (result i32)
+  (global.set $g1
+   (i32.const 100)
+  )
+  (drop
+   (call $no
+    (i32.const 200)
+   )
+  )
+  (global.get $g1)
+ )
+ (func $no2 (; 1 ;) (type $FUNCSIG$ii) (param $x i32) (result i32)
+  (global.set $g1
+   (i32.const 100)
+  )
+  (global.set $g1
+   (local.get $x)
+  )
+  (global.get $g1)
+ )
+ (func $yes (; 2 ;) (type $FUNCSIG$ii) (param $x i32) (result i32)
+  (global.set $g1
+   (i32.const 100)
+  )
+  (global.set $g2
+   (local.get $x)
+  )
+  (i32.const 100)
```

Comment:
Looks like this still got optimized.

Reply:
Correct, it does the optimization itself, but doesn't also run general optimization passes as a followup to that optimization.